src/utils/getTokenPriceInUSD.ts

```diff
@@ -0,0 +1,175 @@
+import { ethers } from "ethers";
+import { getRpcUrl } from "./getRpcUrl.js";
+import {
+  ChainId,
+  currenciesByNetwork,
+  Currency,
+} from "@hypercerts-org/marketplace-sdk";
+
+export const getTokenPriceInUSD = async (
+  chainId: ChainId,
+  tokenAddress: string,
+) => {
+  const provider = new ethers.JsonRpcProvider(getRpcUrl(chainId));
+  // This constant describes the ABI interface of the contract, which will provide the price of ETH
+  // It looks like a lot, and it is, but this information is generated when we compile the contract
+  // We need to let ethers know how to interact with this contract.
+  const aggregatorV3InterfaceABI = [
```

Comment:
don't need the full abi, you can provide the fields for the call you're actually executing. Generally ABIs are stored as JSON files for availability in the whole app as a resource

Reply:
Moved it to a json file. I don't see the benefits of leaving out parts of the ABI, in case we need them at some other point in time.